doc(install): support helm-controller (#1007)
longhorn/website: content/docs/1.8.0/deploy/install/install-with-helm-controller.md

---
title: Install with Helm Controller
weight: 10
---

In this section, you will learn how to install Longhorn with the HelmChart controller built into RKE2 and K3S.

### Prerequisites

- Kubernetes cluster: Ensure that each node fulfills the [installation requirements](../#installation-requirements).  Cluster should be running RKE2 or K3S.

> [This script](https://github.com/longhorn/longhorn/blob/v{{< current-version >}}/scripts/environment_check.sh) can be used to check the Longhorn environment for potential issues.

### Installing Longhorn

> **Note**:

> > * The initial settings for Longhorn can be found in [customized using Helm options or by editing the deployment configuration file.](../../../advanced-resources/deploy/customizing-default-settings/#using-helm)

Reviewer: ```suggestion
> * The initial settings for Longhorn can be [customized using Helm options or by editing the deployment configuration file.](../../../advanced-resources/deploy/customizing-default-settings/#using-helm)
```

> * For Kubernetes < v1.25, if your cluster still enables Pod Security Policy admission controller, set the helm value `enablePSP` to `true` to install `longhorn-psp` PodSecurityPolicy resource which allows privileged Longhorn pods to start.

> 1. Create a HelmChart yaml file something like this:

Reviewer: ```suggestion
1. Create a HelmChart yaml file similar to this:
```

    ```yaml
   apiVersion: helm.cattle.io/v1
    kind: HelmChart
    metadata:
      annotations:
        helmcharts.cattle.io/managed-by: helm-controller
      finalizers:
      - wrangler.cattle.io/on-helm-chart-remove
      generation: 1
      name: longhorn-install
      namespace: default
    spec:

>       version: v1.8.0

Reviewer 2: Replace with a version variable. Same for other versions.

Author: Done, thanks.

      chart: longhorn
      repo: https://charts.longhorn.io
      failurePolicy: abort
      targetNamespace: longhorn-system
      createNamespace: true

    ```

>     > **IMPORTANT!** Ensure that `spec.failurePolicy` is set to "abort".  The only other value is the default: "reinstall", which will do an uninstall of Longhorn.  With "abort", it will simply retry periodically, giving the user a chance to fix the problem.

Reviewer: ```suggestion
    > **IMPORTANT!** Ensure that `spec.failurePolicy` is set to "abort".  The only other value is the default: "reinstall", which performs an uninstall of Longhorn.  With "abort", it retries periodically, giving the user a chance to fix the problem.
```

    instance-manager-77d11dda6091967f9b30011c9876341b   1/1     Running   0             29m
    instance-manager-870c250b69a4fe01382ed46156d33f47   1/1     Running   0             29m
    instance-manager-a4099c5ce28b423c3cc2667906f4b0b4   1/1     Running   0             29m
    longhorn-csi-plugin-jfbh5                           3/3     Running   0             29m
    longhorn-csi-plugin-w768w                           3/3     Running   0             29m
    longhorn-csi-plugin-xcghm                           3/3     Running   0             29m
    longhorn-driver-deployer-586bc86bf9-bkwk6           1/1     Running   0             30m
    longhorn-manager-c4xtv                              1/1     Running   1 (30m ago)   30m
    longhorn-manager-kgqts                              1/1     Running   0             30m
    longhorn-manager-n8xdr                              1/1     Running   0             30m
    longhorn-ui-69667f9678-2lvxn                        1/1     Running   0             30m
    longhorn-ui-69667f9678-2xmc9                        1/1     Running   0             30m

    ```

> 5. To enable access to the Longhorn UI, you will need to set up an Ingress controller. Authentication to the Longhorn UI is not enabled by default. For information on creating an NGINX Ingress controller with basic authentication, refer to [this section.](../../accessing-the-ui/longhorn-ingress)

Reviewer: ```suggestion
5. To enable access to the Longhorn UI, you need to set up an Ingress controller. Authentication to the Longhorn UI is not enabled by default. For information on creating an NGINX Ingress controller with basic authentication, refer to [this section.](../../accessing-the-ui/longhorn-ingress)
```
